packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -664,12 +707,13 @@ export class AppSyncModelVisitor<
       let intermediateModel = this.generateIntermediateModel(
         value[0].model,
         value[1].model,
-        value[0].field,
-        value[1].field,
         graphqlName(toUpper(value[0].directive.arguments.relationName)),
       );
       const modelDirective = intermediateModel.directives.find(directive => directive.name === 'model');
       if (modelDirective) {
+        // Maps @primaryKey and @index of intermediate model to old @key
```

Comment:
Just a comment on if and why the order of processing these directives has significance would be helpful. For example, `processIndex(intermediateModel) then processPrimaryKey(intermediateModel)`

Reply:
> We are adding additional fields (corresponding to sort keys if specified) to the intermediate model. Won't this also change the frontend intermediate type generated by Modelgen? This might impact existing customers (Studio) if it does as I don't see a FF added in this PR.

Just confirm with PM and Datastore engineers that this will not impact studio and datastore customers